File: solutions/ruby/difference-of-squares/1/difference_of_squares.rb
class Squares
  def initialize(num)
    @num = num
  end

  def square_of_sum
    ((1..@num).to_a.sum)**2
  end

  def sum_of_squares
    (1..@num).map { |k| k**2 }.sum
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {

int difference(int largest){
    int result = 0;
    for(int i=1;i<=largest;i++){
        for(int j=1;j<i;j++){
            result += i*j;
        }
    }
    return 2*result;
}

int square_of_sum(int largest){
    int result = 0;
    for(int i=1;i<=largest;i++){
        result += i;
    }
    return result*result;
}
int sum_of_squares(int largest){
    int result = 0;
    for(int i=1;i<=largest;i++){
        result += i*i;
    }
    return result;
}
}  // namespace difference_of_squares
